src/libstore/globals.cc

```diff
@@ -142,12 +142,24 @@ void MaxBuildJobsSetting::set(const std::string & str)
 void initPlugins()
 {
     for (const auto & pluginFile : settings.pluginFiles.get()) {
-        /* handle is purposefully leaked as there may be state in the
-           DSO needed by the action of the plugin. */
-        void *handle =
-            dlopen(pluginFile.c_str(), RTLD_LAZY | RTLD_LOCAL);
-        if (!handle)
-            throw Error(format("could not dynamically open plugin file '%1%': %2%") % pluginFile % dlerror());
+        Paths pluginFiles;
+        try {
+            auto ents = readDirectory(pluginFile);
+            for (auto & ent : ents)
+                pluginFiles.emplace_back(pluginFile + std::move(ent.name));
```

Comment:
Does this move actually accomplish anything? Isn't it a const argument to `+` anyway?

Reply:
Erm, right, this was from before I remembered the `ent` was relative. Will fix.